minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/DockerBuildTask.java

```diff
@@ -0,0 +1,141 @@
+import com.google.cloud.tools.minikube.util.MinikubeDockerEnvParser;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.provider.PropertyState;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+
+/** Task to build Docker images. */
+public class DockerBuildTask extends DefaultTask {
+
+  /** minikube executable : lazily evaluated from extension input */
+  private PropertyState<String> minikube;
+  /** docker executable : lazily evaluated from extension input */
+  private PropertyState<String> docker;
+  /** The set of files to build (PATH | URL | -) */
+  private String context;
+  /** Flag passthrough */
+  private String[] flags = {};
+
+  public DockerBuildTask() {
+    minikube = getProject().property(String.class);
+    docker = getProject().property(String.class);
+  }
+
+  // @VisibleForTesting
+  class CommandExecutorFactory {
+    CommandExecutor createCommandExecutor() {
+      return new CommandExecutor().setLogger(getLogger());
+    }
+  }
+
+  // @VisibleForTesting
+  DockerBuildTask setCommandExecutorFactory(CommandExecutorFactory commandExecutorFactory) {
+    this.commandExecutorFactory = commandExecutorFactory;
+    return this;
+  }
+
+  private CommandExecutorFactory commandExecutorFactory = new CommandExecutorFactory();
+
+  @Input
+  public String getMinikube() {
+    return minikube.get();
+  }
+
+  public void setMinikube(String minikube) {
+    this.minikube.set(minikube);
+  }
+
+  public void setMinikube(PropertyState<String> minikube) {
+    this.minikube = minikube;
+  }
+
+  @Input
+  public String getDocker() {
+    return docker.get();
+  }
+
+  public void setDocker(String docker) {
+    this.docker.set(docker);
+  }
+
+  public void setDocker(PropertyState<String> docker) {
+    this.docker = docker;
+  }
+
+  @Input
+  public String getContext() {
+    return context;
+  }
+
+  public void setContext(String context) {
+    this.context = context;
+  }
+
+  @Input
+  public String[] getFlags() {
+    return flags;
+  }
+
+  public void setFlags(String[] flags) {
+    this.flags = flags;
+  }
+
+  @TaskAction
+  public void execDockerBuild() throws IOException, InterruptedException {
+    // Gets the minikube docker environment variables by running the command 'minikube docker-env'.
+    List<String> minikubeDockerEnvCommand =
+        Arrays.asList(minikube.get(), "docker-env --shell none");
+    List<String> dockerEnv =
+        commandExecutorFactory.createCommandExecutor().run(minikubeDockerEnvCommand);
+
+    Map<String, String> environment;
+    try {
+      environment = MinikubeDockerEnvParser.parse(dockerEnv);
+    } catch (IllegalArgumentException ex) {
+      throw new GradleException("Exception while processing minikube's Docker environment");
```

Comment:
can you add ex to the context?
new GradleException(message, ex)?

Reply:
you might be able to just let this runtime exception fall through and gradle auto converts it.

Reply:
I just looked at the message that the exception in minikubeDockerEnv parser throws, it might not be as helpful as what you've done here. Maybe we can update that message to be more informative rather than the generic messaging it currently outputs (since the class already names itself as though it is parsing minikube env)

https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/blob/master/minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/util/MinikubeDockerEnvParser.java#L42

Reply:
Okay, made the exception fall through and changed the error message in the parser to be more informative.